salt/cloud/clouds/ec2.py

```diff
@@ -1789,7 +1789,7 @@ def request_instance(vm_=None, call=None):
 
     if userdata is not None:
         try:
-            params[spot_prefix + 'UserData'] = base64.b64encode(userdata)
+            params[spot_prefix + 'UserData'] = base64.b64encode(userdata.encode('utf-8'))
```

Comment:
Can you move this to the `oxygen` branch, and use 

`salt.utils.data.encode()`

To encode it instead of `.encode('utf-8')`

Reply:
Based on a conversation with @garethgreenaway on #45624.  I'm planning on closing the issue and this PR. The work done on #45271 tackles this issue more completely.